byterun/caml/misc.h

```diff
@@ -94,13 +94,33 @@ extern caml_timing_hook caml_major_slice_begin_hook, caml_major_slice_end_hook;
 extern caml_timing_hook caml_minor_gc_begin_hook, caml_minor_gc_end_hook;
 extern caml_timing_hook caml_finalise_begin_hook, caml_finalise_end_hook;
 
+/* Windows Unicode support (rest below - char_os is needed earlier) */
+
+#ifdef _WIN32
+typedef wchar_t char_os;
+#else
+typedef char char_os;
+#endif
+
 /* Assertions */
 
 #ifdef DEBUG
+
+#ifdef UNICODE
+/* See https://msdn.microsoft.com/ja-jp/library/b0084kay(v=vs.71).aspx
+   It's not clear why this isn't so obviously documented, as it doesn't
+   seem to have been superseded by a more sensible mechanism! */
+#define WIDEN2(x) L##x
+#define WIDEN(x) WIDEN2(x)
+#define __OSFILE__ WIDEN(__FILE__)
```

Comment:
Could we use macro names less "obvious" than `WIDEN` and `WIDEN2`?  All OCaml-C stub files include `<caml/misc.h>`, directly or indirectly, hence I'm worried about macro name space pollution.  `CAML_WIDEN_STRING_LITERAL{,2}` perhaps?

Reply:
Sorry for the lag - good suggestion, which I've followed.